src/Elastic.Apm/Metrics/MetricsProvider/GcMetricsProvider.cs

```diff
@@ -68,54 +58,18 @@ public GcMetricsProvider(IApmLogger logger, IReadOnlyList<WildcardMatcher> disab
 			_collectGcGen1Size = !WildcardMatcher.IsAnyMatch(disabledMetrics, GcGen1SizeName);
 			_collectGcGen2Size = !WildcardMatcher.IsAnyMatch(disabledMetrics, GcGen2SizeName);
 			_collectGcGen3Size = !WildcardMatcher.IsAnyMatch(disabledMetrics, GcGen3SizeName);
-
-			if (!IsEnabled(disabledMetrics))
-				return;
+			_isEnabled = _collectGcCount || _collectGcTime || _collectGcGen0Size || _collectGcGen1Size || _collectGcGen2Size || _collectGcGen3Size;
+			if (!IsEnabled(disabledMetrics)) return;
 
 			_logger = logger.Scoped(DbgName);
-
-			if (PlatformDetection.IsDotNetFullFramework)
+			if (!PlatformDetection.IsDotNetCore && !PlatformDetection.IsDotNet)
 			{
-				// if the OS doesn't support filtering, the processes and runtimes tracked by the trace event session source can grow
-				// unbounded, leading to the application consuming ever more memory.
-				if (!TraceEventProviderOptions.FilteringSupported)
-				{
-					_logger.Info()?.Log("TraceEventSession does not support filtering on this operating system. GC metrics won't be collected");
-					return;
-				}
-
-				try
-				{
-					TraceEventSessionName = SessionNamePrefix + Guid.NewGuid();
-					_traceEventSession = new TraceEventSession(TraceEventSessionName);
-					_traceEventSession.Source.NeedLoadedDotNetRuntimes();
-					_traceEventSession.EnableProvider(
-						ClrTraceEventParser.ProviderGuid,
-						TraceEventLevel.Informational,
-						(ulong)ClrTraceEventParser.Keywords.GC, // garbage collector details
-						new TraceEventProviderOptions { ProcessIDFilter = new List<int>(1) { Process.GetCurrentProcess().Id } }
-					);
-
-					_traceEventSession.Source.Clr.GCHeapStats += ClrOnGCHeapStats;
-					_traceEventSession.Source.AddCallbackOnProcessStart(process =>
-					{
-						process.AddCallbackOnDotNetRuntimeLoad(runtime =>
-						{
-							runtime.GCEnd += RuntimeGCEnd;
-						});
-					});
-
-					_traceEventSessionTask = Task.Run(() => _traceEventSession.Source.Process());
-				}
-				catch (Exception e)
-				{
-					_logger.Warning()?.LogException(e, "TraceEventSession initialization failed - GC metrics won't be collected");
-					return;
-				}
+				_logger.Info()?.Log("Gc metrics are only available on .NET Core, disabling metric collection");
```

Comment:
```suggestion
				_logger.Info()?.Log("GC metrics are only available on .NET Core, disabling metric collection");
```

Naming makes this so hard... We'd print `...only available on .NET Core...`, but we actually mean .NET Core and .NET5+. And over time due to the naming change .NET Core is less relevant, and .NET X will be the only supported one. I don't know how to address this. Should we maybe print `.NET Core and .NET 5+` ?